### ISSUE TITLE
`True & True` is considered to be `int`

### ISSUE THREAD

**Issue by lancelote** [astral-sh/ty#270]
### Summary

given ...

```python
from typing import reveal_type

reveal_type(True & True)
```

`ty` reports ...

```
% ty check sample.py 
info: revealed-type: Revealed type
 --> sample.py:3:1
  |
1 | from typing import reveal_type
2 |
3 | reveal_type(True & True)
  | ^^^^^^^^^^^^^^^^^^^^^^^^ `int`
  |

Found 1 diagnostic
```

### Version

ty 0.0.0-alpha.7 (905a3e1e5 2025-05-07)

**Comment by AlexWaygood** [comment]
Amusingly, we give a worse result here for the literal `True` constants than we do if all the information you tell us is that there are two instances of `bool`:

```py
from typing import reveal_type

reveal_type(True & True)  # revealed: bool

def f(x: bool, y: bool):
    reveal_type(x & y)  # revealed: int
```

https://play.ty.dev/1cfad6de-0386-4733-84f9-df9c1c7e7327

**Comment by lancelote** [comment]
My "real world" example was something like that ...

```python
from typing import Any, reveal_type


class SomeClass:
    def some_func(self, param: Any) -> None:
        flag = True

        ...

        flag &= param is self
        reveal_type(flag)  # `ty` says `int`
```

**Comment by lancelote** [comment]
@AlexWaygood Thank you for an instant fix 👍
